### PR TITLE
Add avifclaptest.cc for testing 'clap' properties

### PR DESCRIPTION
**Pull request by wantehchang**
This file has a test for the avifCropRectConvertCleanApertureBox() function. The test was ported from the test deleted in patchset 9 of https://chromium-review.googlesource.com/c/chromium/src/+/4585301/8..9. I wrote that test originally.

More tests related to the 'clap' (clean aperture) image properties can be added to this file in the future.

### CODE REVIEW

**Review by y-guyon — APPROVED** [review]
Thank you for adding this test. It requires more cases of course but this is a good start.

As obvious as it sounds:
- Why is all of the `avifCleanApertureBox` API using `uint32_t` if part of it is converted to `int32_t` internally?
- Why is there `widthN` and `widthD` if only integers are accepted in the end?

File: tests/gtest/avifclaptest.cc [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  avifPixelFormat yuv_format;
+  avifCleanApertureBox clap;
+
+  bool expected_result;
+  avifCropRect expected_crop_rect;  // Ignored if expected_result is false.
+};
+
+AVIFClapPropertyParam kAVIFClapPropertyTestParams[] = {
+    //////////////////////////////////////////////////
+    // Negative tests:
```
[the remark this answers is not in the document]
Comment (wantehchang) [reply tests/gtest/avifclaptest.cc]: Thanks for the suggestions. I actually attempted your first suggestion last Friday, but perhaps in a rush to wrap this PR up before the weekend, I didn't finish it. I just finished it now.

I am afraid that I don't understand your second suggestion of replacing the parameterized `TEST_P` with reguar `TEST` cases. Did you mean `TEST_P` should only be used with `testing::Combine()`?

**Review by wantehchang — COMMENTED** [review]
Yannis: Thanks for the review. I addressed your comment. Please review the new commit.

File: tests/gtest/avifclaptest.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (y-guyon) [reply tests/gtest/avifclaptest.cc]: > I am afraid that I don't understand your second suggestion of replacing the parameterized `TEST_P` with reguar `TEST` cases. Did you mean `TEST_P` should only be used with `testing::Combine()`?

It just has little interest over simpler approaches, such as

```c
void ValidateClapProperty(
  uint32_t width, uint32_t height, avifPixelFormat yuv_format, avifCleanApertureBox clap, avifCropRect expected_crop_rect) {
  avifCropRect crop_rect;
  avifDiagnostics diag;
  EXPECT_TRUE(avifCropRectConvertCleanApertureBox(&crop_rect, &clap, width, height, yuv_format, &diag));
  EXPECT_EQ(crop_rect.x, expected_crop_rect.x);
  EXPECT_EQ(crop_rect.y, expected_crop_rect.y);
  EXPECT_EQ(crop_rect.width, expected_crop_rect.width);
  EXPECT_EQ(crop_rect.height, expected_crop_rect.height);
}

TEST(ValidClapPropertyTest, ValidClapProperty) {
    // pcX = 0 + (120 - 1)/2 = 59.5
    // pcY = 0 + (160 - 1)/2 = 79.5
    // leftmost = 59.5 - (96 - 1)/2 = 12
    // topmost = 79.5 - (132 - 1)/2 = 14
    ValidateClapProperty(120, 160, AVIF_PIXEL_FORMAT_YUV420, {96, 1, 132, 1, 0, 1, 0, 1}, {12, 14, 96, 132});
    // pcX = -30 + (120 - 1)/2 = 29.5
    // pcY = -40 + (160 - 1)/2 = 39.5
    // leftmost = 29.5 - (60 - 1)/2 = 0
    // topmost = 39.5 - (80 - 1)/2 = 0
    ValidateClapProperty(120, 160, AVIF_PIXEL_FORMAT_YUV420, {60, 1, 80, 1, static_cast<uint32_t>(-30), 1, static_cast<uint32_t>(-40), 1}, {0, 0, 60, 80});
};
```

but it does not matter much and it is not worth spending too much time over it. Adding tests is already great, thanks!

**Review by y-guyon — APPROVED** [review]

**Comment by y-guyon** [comment]
> Given the above, perhaps we can consider it a bug that these fields are not declared as `signed int(32)`, and change their type to `int32_t` in libavif?

Yes. Maybe we do not have to reproduce the specifications exactly in avif.h, and keep a convenient API for the use cases related to AVIF?

For example it could be

```c
typedef struct avifCleanApertureBox
{
    // Adaptation of 'clap' from ISO/IEC 14496-12:2015 12.1.4.3
    uint32_t width;
    uint32_t height;
    int32_t horizOffN;
    uint32_t horizOffD;
    int32_t vertOffN;
    uint32_t vertOffD;
} avifCleanApertureBox;
```

It does not matter much since most users will use the helper function to create it from rectangle corner coordinates. But then exposing `avifCleanApertureBox` at all is not very useful either and just adds an extra step for the user.

Anyway this is not really important and is not worth spending too much time on it. Sorry for the digression from this PR's actual focus.

**Comment by wantehchang** [comment]
Don't worry about the digression. It's good to document what I know about the CleanApertureBox in ISO BMFF.

**Pull request merged**